src/realm/impl/sequential_getter.hpp

```diff
@@ -84,7 +89,10 @@ class SequentialGetter : public SequentialGetterBase {
 #pragma warning(push)
 #pragma warning(disable : 4800) // Disable the Microsoft warning about bool performance issue.
 #endif
+        return m_column->get(index);
 
+        // FIXME: Below optimization is skipped because array leafs might relocate during the lifetime of the 
+        // object that owns this SequentialGetter. Enable again when we have proper support for that.
```

Comment:
If you fix this, also note the unreachable return [above](https://github.com/realm/realm-core/pull/2314/files?diff=unified#diff-6bb1b9ac8faad726a17005f031184aa0R82)